### DISCUSSION TITLE
Site login problem e107 version 1.7

### DISCUSSION THREAD

**Discussion by Taffman** [e107inc/e107#5207] [General]
I've suddenly started to see the following PHP error on user login:

 PHP Fatal error:  Call to undefined function print_a() in /home/site/public_html/class2.php on line 1985

I realise there is no support for the old e107 version 1.7 anymore but I'm hoping someone can help me with this as I'm stuck with no one being able to log into my site and don't know where to go for help. Users get an insecure site warning even though it's an HTTPS site and has worked ok up until now. Users attempting to login get stuck in an endless loop where site is causing browser to think it's an insecure connection and returns the user to the login page.

If anyone here can point me in the right direction I'd be very grateful, I will upgrade to version 2, honest, I have my reasons why I haven't done so up until now.

**Comment by Jimmi08** [comment]
check if your hosting haven't changed PHP version 
that line is used if you set debug mode on

**Reply by Taffman** [reply]
Currently running PHP Version 5.6.40 built in December 23, so not sure if this is the problem as its only started to occur recently?

**Comment by tgtje** [comment]
a) Try to access your database directly and do a search for htt P   (not http  S). Iff a htt P address exists you maybe can change it (dependicity). A present htt P url can cause issues.
b) insecure site warning  quite common thesedays mostly related ( if enabled) to the login function/handling
The present newer browser versions react quite strict .
Do not expect 5.6.4 to be a real issue, but not much info, as most went from 5.4  to php 7 and above.
c) using a ssl certificate ? is it up to date eg maybe renew

**Reply by Taffman** [reply]
Thanks for the advice. I can access the database via phpmyadmin so I'll try a search and replace for http:.

SSL cert is up to date, renewed recently.

Strange things started to happen a few days ago, lost some values from preferences i.e. site tagline, admin contact details etc. Then couldn't update custom pages, noticed in admin that database needed to be updated so I did that and page edits became possible again. Lost access to Chatbox2 config went back to default settings, can't change them in admin. no idea how this has happened.

**Reply by Taffman** [reply]
Just did a dB search for "http:", none found so back to square one.

**Reply by tgtje** [reply]
Going back in the history of closed issues. Helas as it seems to have happened. Older v 1 plugins do have issues with php 5.6.

Let's hope your hosting still has  php 5.4 somewhere (**and allows running**)
Test with adding

SetEnv DEFAULT_PHP_VERSION 54

At the top of .htaccess file. 
Mind : unsure iff problems occur, please backup your database anyhow. REMEMBER : if works = temp. solution / not really adviced.
And reading Chatbox2 : old .. your site will; iff more older unsupported ( by php coding) plugins on board; run in troubles on newer versions.. both on php and e107.
Best advice i can give : setup a home local system create a 'copy' of your site and test (even so you might have issues getting php 5.4 versions as (pc use) windoos will affect that too. When possible:  try updating bit by bit on php and version of e107.
Too much has changed since 2015-16.

**Reply by Taffman** [reply]
Ok many thanks for this I'll try and downgrade PHP to version 5.4.

My database version is 5.5.5-10.2.36-MariaDB any issues with that?

**Reply by tgtje** [reply]
Can not really answer on that (in working) . They (pers.opinion) should/could be newer in terms of security..
please mind : it will likely not disable the browsers warning i'am afraid, even php 8 v 2.3.3 gives them (= login based credential issues/ transmit etc....browsers sees)

**Comment by Taffman** [comment]
OK so I think the PHP error I originally reported was a red herring and a consequence of me being in debug mode at that time.
What I can confirm is that all browsers are refusing to post the login form as they see it as an insecure connection.

my site is an https site, however when I "View Page Source" for login.php I see the following:

<form action='[http://mysite/login.php](view-source:mysite/login.php)' method='post' id='myform'>

so my question is where is http:// coming from, reference to site url is via short codes in login.php so is there somewhere within e107 where I should set this up:

e107 preferences has the site url as https://mysite.org
I have an https redirect set up in .htaccess also as follows:

RewriteCond %{HTTPS} off
RewriteCond %{HTTP_HOST} ^(www\.)?domain\.org$ [NC]
RewriteRule ^(.*)$ http://site/$1 [L,R=301]

If I can get the login form to submit to https://mysite.org instead of http://mysite.org I think my users and I will be able to log into my site once again.

**Reply by Moc** [reply]
> so my question is where is http:// coming from, 

From the .htaccess RewriteRule you posted. 

Closing this discussion as v1.x (let alone 0.7) is no longer supported. Please let us know if you need any help updating to v2.
